### PR TITLE
UserRepo and BillRepo

### PR DESCRIPTION
**Pull request by Mincer2412**
UserRepo added. findById() and update() methods added.

### CODE REVIEW

File: src/main/java/com/epam/repo/UserRepo.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import java.sql.SQLException;
+import java.sql.Statement;
+
+
+public class UserRepo implements Repository<User, Long> {
+
+    Statement statement = null;
+    String sql = "SELECT * FROM USER WHERE ID = ";
+
+    {
```
Comment (IlyasYOY) [inline src/main/java/com/epam/repo/UserRepo.java]: maybe constructor is better place?

File: src/main/java/com/epam/repo/UserRepo.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        try {
+            statement = connection.createStatement();
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+
+    }
+
+    @Override
+    public User save(User item) {
```
Comment (IlyasYOY) [inline src/main/java/com/epam/repo/UserRepo.java]: null?

File: src/main/java/com/epam/repo/UserRepo.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            while (resultSet.next()) {
+                Long tmpId = Long.parseLong(resultSet.getString("ID"));
+
+                if (tmpId.equals(id)) {
+                    user.setId(tmpId);
+                    user.setLogin(resultSet.getString("LOGIN"));
+                    user.setPassword(resultSet.getString("PASSWORD"));
+
+                    String role = resultSet.getString("ROLE");
+                    user.setRole(Enum.valueOf(UserRole.class, role));
```
Comment (IlyasYOY) [inline src/main/java/com/epam/repo/UserRepo.java]: `Role.valueOf`

File: src/main/java/com/epam/repo/UserRepo.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Override
+    public User removeById(Long id) {
+        return null;
+    }
+
+    @Override
+    public User findById(Long id) {
+        ResultSet resultSet = null;
+        User user = User.builder().build();
```
Comment (IlyasYOY) [inline src/main/java/com/epam/repo/UserRepo.java]: Check Alex's PR.

File: src/main/java/com/epam/repo/UserRepoImpl.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @SneakyThrows
+    @Override
+    public User findById(Long id) {
+        ResultSet resultSet = null;
+        User user = null;
+
+        if (id == null) {
+            return null;
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/repo/UserRepoImpl.java]: Why you want to return null here? In Java we have really good solution for method abnormal termination - exceptions. Here, you can use some runtime exceptions, such as IllegalArgument or create your own

**Comment by Mincer2412** [comment]
BillRepoImpl and UserRepoImpl updated. IllegalArgumentException added where needed.
BillRepoImplTest class created.
A few Bill tests in BillRepoImplTest added.
BilService and UserService created and delegate methods from BillRepoImpl and UserRepoImpl respectively.

**Comment by imbirevkotlya** [comment]
Seems great now, please, add the javadocs and unit tests for this

**Pull request merged**